src/heretic/s_sound.c

```diff
@@ -85,6 +85,7 @@ void S_StartSong(int song, boolean loop)
     {
         I_StopSong();
         I_UnRegisterSong(rs);
+        rs = NULL;
```

Comment:
This PR exposed undefined behavior in Heretic's `S_StartSong()` that caused the game to segfault when attempting to show the error message. A small fix has been added here, [similar to Hexen](https://github.com/chocolate-doom/chocolate-doom/blob/e9c202bd4df494634509e6023ad426b93df3871b/src/hexen/s_sound.c#L178).